web/html/src/manager/systems/details/mgr-server-info.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from "react";
+
+import { TopPanel } from "components/panels/TopPanel";
+import { AsyncButton } from "components/buttons";
+
+import { Messages } from "components/messages";
+import { Utils as MessagesUtils } from "components/messages";
+
+import * as Systems from "components/systems/details";
+
+import Network from "utils/network";
+
+// See java/code/src/com/suse/manager/webui/templates/systems/mgr-server.jade
+type Props = {
+  serverId: string;
+  name: string;
+  version: string;
+  reportDbName: string;
+  reportDbHost: string;
+  reportDbPort: number;
+  reportDbUser: string;
+  reportDbLastSynced?: string;
+  isAdmin: boolean;
+  /** Locale of the help links */
+  docsLocale: string;
+};
+
+type State = {
+  messages: any[];
+};
+
+const msgMap = {
+  "Not a system id": t("Not a system id"),
+  "Unknown System": t("Unknown System"),
+  "System is not a Mgr Server": t("System is not a peripheral server"),
+};
+
+class MgrServer extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      messages: [],
+    };
+  }
+
+  handlePwReset() {
+    return Network.post(`/rhn/manager/api/systems/${this.props.serverId}/mgr-server-reportdb-newpw`, { })
```

Comment:
Unless passing this empty object along with the post request is important, I would remove it. If it is important, I would suggest adding a comment why it's needed.

Reply:
I was not aware that you can omit the parameter.